src/transformers/models/mamba2/modeling_mamba2.py

```diff
@@ -450,13 +455,13 @@ def forward(
         hidden_states: torch.FloatTensor,
         attention_mask: torch.FloatTensor,
         position_ids: torch.LongTensor,
-        cache: Optional[HybridMamba2AttentionDynamicCache] = None,
+        cache_params: Optional[HybridMamba2AttentionDynamicCache] = None,
```

Comment:
Sounds good! So we want to change `cache_params` to `past_key_values` in attention related classes/methods and keep it `cache_params` in mixture related ones? And where do you plan to use `cache`?

Reply:
Ok, so it's a bit complicated since we depend on the name "past_key_values" on top-level, i.e. Mamba2Model, ForCausal etc.

`cache` would be for the Mamba2Block class, `cache_params` for the Mamba2Mixer, and `past_key_values` for all attention related ones, e.g. Mamba2Attention.

Reply:
Take a look now, it should make sense. Do you want to also change the parameter from `past_key_values` to `cache` in all models using the Mamba2Model class?